### ISSUE TITLE
Loss Always Not Change

### ISSUE THREAD

**Issue by AugustusHsu** [AugustusHsu/Music-Recommendation-System#1]
Use Softmax on the Output Layer, the loss is only decreased a little
``` # Calculate cross-entropy loss
with tf.name_scope('loss'):
    log_loss = tf.nn.softmax_cross_entropy_with_logits(logits=self.Output, labels=self.onehot_label)
    self.loss = tf.reduce_mean(log_loss)
```

```# Output layer setting
with tf.name_scope('OUTPUT_LAYER'):
    initial_value = tf.truncated_normal([self.FC_output.shape[1].value, self.output_dim],
                                                               stddev=0.1)
    self.output_weight = tf.Variable(initial_value, name='output_weight')
    self.output_bais = tf.Variable(tf.zeros(shape=[self.output_dim]), name='output_bais')
    self.Output = tf.matmul(self.FC_output, self.output_weight) + self.output_bais
    self.Output = tf.nn.softmax(self.Output)
    print('Output')
    print(self.Output.shape)
```

**Comment by AugustusHsu** [comment]
[softmax_cross_entropy_with_logits_v2](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits_v2)：

WARNING: This op expects unscaled logits, since it performs a softmax on logits internally for efficiency. Do not call this op with the output of softmax, as it will produce incorrect results.

[tf.nn.softmax_cross_entropy_with_logits_v2](https://github.com/AugustusHsu/Music-Recommendation-System/blob/d8f4a7ba431820a56a6f7322d1527e9c70b31438/Recommendation_Model.py#L193) can't use the softmax at the [output layer](https://github.com/AugustusHsu/Music-Recommendation-System/blob/d8f4a7ba431820a56a6f7322d1527e9c70b31438/Recommendation_Model.py#L171)
